### ISSUE TITLE
code

### ISSUE THREAD

**Issue by maxingchu** [maxingchu/-#1]
import matplotlib.pyplot as plt
import pandas as pd
from sklearn import linear_model
from sklearn.preprocessing import PolynomialFeatures as PF
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.ensemble import ExtraTreesClassifier
from sklearn.linear_model import LogisticRegression

file_path_face = "D:\\file_beita\\FBLTV21.csv"
file_path_goog = "D:\\file_beita\\11111.csv"

# 读取facebook模型文件，输出X,Y参数
def read_face_file():
    csv_ltv_face = pd.read_csv(file_path_face)
    # print(csv_ltv)

    # regr = linear_model.LinearRegression()
    X_param = []
    Y_param = []
    # 拟合
    # regr.fit(csv_ltv_face['D0'],csv_ltv_face['D7'])
    # 求出直线的斜率
    # a,b = regr.coef_,regr.intercept_
    # 求知
    # plt.scatter(csv_ltv_face['D0'], csv_ltv_face['D7'], color='blue')
    # 2.拟合的直线
    # plt.plot(csv_ltv_face['D0'], regr.predict(csv_ltv_face['D7']), color='red', linewidth=4)

    # plt.show()

    for s1, s2 in zip(csv_ltv_face['0'], csv_ltv_face['1']):
        X_param.append([float(s1)])
        Y_param.append(float(s2))
    predict_Y = 0.2

    print(X_param)
    print(Y_param)
    #线性回归
    result = linear_model_main(X_param, Y_param, predict_Y)
    #岭回归
    result1 = ridge_regr(X_param, Y_param, predict_Y)
    #随机森林
    result2 = ridge_regr(X_param,Y_param,predict_Y)
    print("Intercept_value", result['intercept'])
    print("coefficient", result['coefficient'])
    print("predicted_value", result['predict_Y'])
    print('predicted_left', result['predict_left'])
    print('pedicted_right', result['predict_right'])
    print("predicted_value1", result1['predict_Y'])
    print('predicted_left1', result1['predict_left'])
    print('pedicted_right1', result1['predict_right'])
    print("predicted_value2", result2['predict_Y'])
    show_linear_line(X_param, Y_param)
    # print(X_param)
    # regr.fit(X_param, Y_param)
    # plt.scatter(X_param, X_param, color='blue')
    # plt.plot(X_param, X_param, color='red')
    # plt.xticks(())
    # plt.yticks(())
    # plt.show()

# 线性回归训练模型
def linear_model_main(X_param, Y_param, predict_Y):
    # Create linear regression object
    regr = linear_model.LinearRegression()
    regr.fit(X_param, Y_param)
    predict_out = regr.predict(predict_Y)
    predictions = {}
    predictions['intercept'] = regr.intercept_
    predictions['coefficient'] = regr.coef_
    predictions['predict_Y'] = predict_out
    predictions['predict_left'] = predict_out - [0.008]
    predictions['predict_right'] = predict_out + [0.008]
    return predictions

# 线性回归分析图
def show_linear_line(X_param, Y_param):
    # Create linear regression object
    regr = linear_model.LinearRegression()
    regr.fit(X_param, Y_param)
    plt.scatter(X_param, Y_param, color='blue')
    plt.plot(X_param, regr.predict(X_param), color='red', linewidth=2)
    plt.xticks(())
    plt.yticks(())
    plt.show()

# 多项式处理预测曲线
def poly_regre():
    file_goog = pd.read_csv(file_path_goog)
    X = file_goog.iloc[:, 2:3].values
    Y = file_goog.iloc[:, 3].values
    print(X)
    print(Y)
    PF_regr = PF(degree=4)
    X_poly = PF_regr.fit_transform(X)
    lin_reg_2 = linear_model.LinearRegression()
    lin_reg_2.fit(X_poly, Y)
    # picture
    plt.scatter(X, Y, color='red')
    plt.plot(X, lin_reg_2.predict(PF_regr.fit_transform(X)), color='blue')
    plt.title("Truth or Bluff(PF)")
    plt.xlabel("Position D0")
    plt.ylabel("Position D7")
    plt.show()

# 岭回归模型
def ridge_regr(X_param, Y_param, predict_Y):
    # file_path_face = pd.read_csv(file_path_goog)
    clf = linear_model.RidgeCV(alphas=[0.01, 1.0, 10.0], cv=None, fit_intercept=True, scoring=None, normalize=False)
    clf.fit(X_param, Y_param)
    predict_out = clf.predict(predict_Y)
    predictions = {}
    predictions['intercept'] = clf.intercept_
    predictions['coefficient'] = clf.coef_
    predictions['predict_Y'] = predict_out
    predictions['predict_left'] = predict_out - [0.008]
    predictions['predict_right'] = predict_out + [0.008]
    return predictions

# 随机森林模型
def ran_for(X_param, Y_param, predict_y):
    print('start train')
    clf2 = RandomForestClassifier(n_estimators=100)
    clf2.fit(X_param,Y_param)
    print(clf2.classes_)
    result = clf2.predict_proba(predict_y)
    return  result
#goog数据预测
def read_goog_file():
    csv_ltv_goog = pd.read_csv(file_path_goog)
    print(csv_ltv_goog)
    X_param = []
    Y_param = []
    for s1, s2 in zip(csv_ltv_goog['D0'], csv_ltv_goog['D7']):
        X_param.append([float(s1)])
        Y_param.append(float(s2))
    predict_Y = 0.89
    print(X_param)
    print(Y_param)
    result = linear_model_main(X_param, Y_param, predict_Y)
    print("Intercept_value", result['intercept'])
    print("coefficient", result['coefficient'])
    print("predicted_value", result['predict_Y'])
    print('predicted_left', result['predict_left'])
    print('pedicted_right', result['predict_right'])
    show_linear_line(X_param, Y_param)

if __name__ == '__main__':
    read_face_file()
    # read_goog_file()
    # poly_regre()
    # read_test()

**Comment by maxingchu** [comment]
bye
